test_vm/tests/market_miner_withdrawal_test.rs

```diff
@@ -0,0 +1,262 @@
+use fil_actor_market::Method as MarketMethod;
+use fil_actor_market::WithdrawBalanceParams as MarketWithdrawBalanceParams;
+use fil_actor_miner::Method as MinerMethod;
+use fil_actor_miner::WithdrawBalanceParams as MinerWithdrawBalanceParams;
+use fil_actor_power::{CreateMinerParams, CreateMinerReturn, Method as PowerMethod};
+use fil_actors_runtime::test_utils::{MARKET_ACTOR_CODE_ID, MINER_ACTOR_CODE_ID};
+use fil_actors_runtime::{STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR};
+use fvm_ipld_blockstore::MemoryBlockstore;
+use fvm_ipld_encoding::BytesDe;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::address::Address;
+use fvm_shared::bigint::bigint_ser::BigIntDe;
+use fvm_shared::bigint::BigInt;
+use fvm_shared::bigint::Zero;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::error::ExitCode;
+use fvm_shared::sector::RegisteredPoStProof;
+use fvm_shared::METHOD_SEND;
+use test_vm::util::{apply_ok, create_accounts};
+use test_vm::Actor;
+use test_vm::VM;
+
+#[cfg(test)]
+mod market_tests {
+    use super::*;
+
+    #[test]
+    fn withdraw_all_funds() {
+        let store = MemoryBlockstore::new();
+        let (v, caller) = market_setup(&store);
+
+        let three_fil = TokenAmount::from(3);
+        assert_add_collateral_and_withdraw(
+            &v,
+            three_fil.clone(),
+            three_fil.clone(),
+            three_fil,
+            *STORAGE_MARKET_ACTOR_ADDR,
+            caller,
+        );
+    }
+
+    #[test]
+    fn withdraw_as_much_as_possible() {
+        let store = MemoryBlockstore::new();
+        let (v, caller) = market_setup(&store);
+
+        // Add 2 FIL of collateral and attempt to withdraw 3
+        let two_fil = TokenAmount::from(2);
```

Comment:
All of the fil accounting here is off by 10^18.  TokenAmount is 1*10^18 FIL (1 attoFIL) all of these numbers need to be multiplied by 10^18

Reply:
Corrected all occurrences of FIL.